src/game.cpp

```diff
@@ -5742,6 +5758,13 @@ int game::assign_faction_id()
     return ret;
 }
 
+long game::assign_creature_id()
+{
+    long ret = next_creature_id;
```

Comment:
Quick math, the frontier the player can reveal is roughly 130x2 squares per turn, or 260 squares.  This assumes a diagonal step per turn, but neglects above and below ground since they're very sparse on squares creatures can spawn on.
With an arbitrary assumption of 1/100 squares containing a new creature, we have ~3 creatures per turn, dividing available long numbers by 3 gives us 715827882 turns before we wrap, dividing that by 10 * 60 * 60 * 24 gives us 828 days of generating 3 creatures per turn before a wrap.  Keep in mind that a number of monsters are going to recycle, including all dynamically spawned monsters (mostly animals), and I intend to de-spawn horde members at some point to cut down on pointless state.

That having been said, the numbers could wrap after nearly three years of constant gameplay, and in reality the average monster spawn rate is going to be WAY lower than 3/turn, so yea, it should be ok unless we add something that massively increases spawn rate.

Side note, it's going to wrap to a negative number and possibly totally break something instead of just wrapping.

I'd still like it to issue a debug message when it wraps, and handle wrapping to a valid number gracefully.